### PR TITLE
Feature/release wa4 rebased

### PR DESCRIPTION
**Pull request by MateiHMCTS**
### Change description ###
Rebased release-wa4 onto release3-wa3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/work-allocation/components/task-list-filter/task-list-filter.component.ts [hunk 1]
```diff
@@ -251,8 +251,11 @@ export class TaskListFilterComponent implements OnInit, OnDestroy {
     };
     let baseLocation = null;
     // if there are no booking locations selected then check for base location for salary judge
-    if ((locations.length === 0) && this.route.snapshot.data && this.route.snapshot.data.locations) {
-      baseLocation = this.route.snapshot.data.locations;
+    if ((locations.length === 0) && this.route.snapshot.data && this.route.snapshot.data.location) {
+      const location: Location = this.route.snapshot.data.location;
+      if (location) {
+        baseLocation = [location];
+      }
```
Comment (connorpgpmcelroy) [inline src/work-allocation/components/task-list-filter/task-list-filter.component.ts:258]: I believe the old code is what we want as the locations are being given in an array

Comment (connorpgpmcelroy) [reply src/work-allocation/components/task-list-filter/task-list-filter.component.ts:258]: At least that is what Kartheek told me

File: src/work-allocation/containers/all-work-task/all-work-task.component.spec.ts [hunk 2]
```diff
@@ -15,10 +15,9 @@ import { WorkAllocationComponentsModule } from '../../components/work-allocation
 import { FieldConfig } from '../../models/common';
 import { Task } from '../../models/tasks';
 import { CaseworkerDataService, LocationDataService, WASupportedJurisdictionsService, WorkAllocationFeatureService, WorkAllocationTaskService } from '../../services';
-import { getMockCaseRoles, getMockTasks } from '../../tests/utils.spec';
+import { getMockTasks } from '../../tests/utils.spec';
 import { AllWorkTaskComponent } from './all-work-task.component';
 import { AllocateRoleService } from 'src/role-access/services';
```
Comment (connorpgpmcelroy) [inline src/work-allocation/containers/all-work-task/all-work-task.component.spec.ts:20]: Need to remove and replace src

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Some minor changes requested

**Review by connorpgpmcelroy — APPROVED** [review]
Comments resolved - all looks good

**Review by Surekha-HMCTS — APPROVED** [review]

**Pull request merged**